Amazon.QLDB.Driver.Tests/AsyncTransactionExecutorTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.QLDB.Driver.Tests
+{
+    using System.Threading.Tasks;
+    using System.Collections.Generic;
+    using Amazon.IonDotnet.Tree;
+    using Amazon.IonDotnet.Tree.Impl;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Moq;
+
+    [TestClass]
+    public class AsyncTransactionExecutorTests
+    {
+        private static AsyncTransactionExecutor asyncTransactionExecutor;
+        private static readonly string query = "my query";
+        private static Mock<IAsyncResult> mockResult;
+        private static Mock<AsyncTransaction> mockTransaction;
+
+        [ClassInitialize]
+#pragma warning disable IDE0060 // Remove unused parameter
+        public static void Setup(TestContext context)
+#pragma warning restore IDE0060 // Remove unused parameter
+        {
+            mockResult = new Mock<IAsyncResult>();
+            mockTransaction = new Mock<AsyncTransaction>(It.IsAny<Session>(), It.IsAny<string>(), null, null);
+            mockTransaction.Setup(t => t.Abort()).Returns(Task.CompletedTask);
+            mockTransaction.Setup(t => t.Execute(It.IsAny<string>())).ReturnsAsync(mockResult.Object);
+            mockTransaction.Setup(t => t.Execute(It.IsAny<string>(), It.IsAny<List<IIonValue>>()))
+                .ReturnsAsync(mockResult.Object);
+            mockTransaction.Setup(t => t.Execute(It.IsAny<string>(), It.IsAny<IIonValue[]>()))
+                .ReturnsAsync(mockResult.Object);
+        }
+
+        [TestInitialize]
+        public void SetupTest()
+        {
+            asyncTransactionExecutor = new AsyncTransactionExecutor(mockTransaction.Object);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(TransactionAbortedException), AllowDerivedTypes = false)]
+        public async Task TestAbortThrowsAbortException()
```

Comment:
Lets add Async to the name to distinguish from the non-async version. ie. `TestAsyncAbortThrowsAbortException`.

Likewise with the other tests.

Reply:
Done